### ISSUE TITLE
Illegal instruction with tensorflow

### ISSUE THREAD

**Issue by rlsharp** [MycroftAI/mycroft-precise#44]
I am installing mycroft-precise on ubuntu on ARCH=x86-64 machine. I followed all instructions for installing source. All seemed to install fine. I then followed the instruction to train and used precise-collect and put the samples the in directories described (except I used dif phrase then hey-computer). I then ran precise-train and I get "Using TensorFlow backend. Illegal instruction with tensorflow"

Any ideas?

**Comment by MatthewScholefield** [comment]
Unfortunately this is due to tensorflow dropping support for old CPUs (ones without AVX support). You can still see if it works by installing an older version with:
```
source .venv/bin/activate
pip install tensorflow==1.5.0
```
Note: This might make models from newer version incompatible. Another option is to compile tensorflow for your cpu from source.

**Comment by rlsharp** [comment]
Thanks Matthew for taking the time to help me. Indeed that seems to be the problem. I did have to
reinstall precise and change the requirements to tensorflow for 1.5 but it worked after that. It did not work well but I expect some is due to working with the earlier version when not expected. The install script should check to see if the CPU supports AVX. Again thanks.

**Comment by MatthewScholefield** [comment]
Good idea. Looks like I can do something like `grep avx /proc/cpuinfo` to check for it. Can you paste the output of `cat /proc/cpuinfo` to make sure that's the issue?

Thanks

**Comment by rlsharp** [comment]
Looks like a good idea. I did a "cat /proc/cpuinfo | grep avx" and got nothing then did the same thing on a aws server I use and got avx in flags lines. Anyway, below is the result of the cat /proc/cpuinfo on my server you requested.  Note, to minimize space used here I only list the results for the first processor since the info is just repeated for the other 7 cores of my machine.

Ron
----------------------
processor	: 0
vendor_id	: GenuineIntel
cpu family	: 6
model		: 26
model name	: Intel(R) Xeon(R) CPU           E5507  @ 2.27GHz
stepping	: 5
microcode	: 0x1d
cpu MHz		: 2261.166
cache size	: 4096 KB
physical id	: 1
siblings	: 4
core id		: 0
cpu cores	: 4
apicid		: 16
initial apicid	: 16
fpu		: yes
fpu_exception	: yes
cpuid level	: 11
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm dca sse4_1 sse4_2 popcnt lahf_lm ssbd ibrs ibpb stibp kaiser tpr_shadow vnmi flexpriority ept vpid dtherm flush_l1d
bugs		: cpu_meltdown spectre_v1 spectre_v2 spec_store_bypass l1tf
bogomips	: 4522.33
clflush size	: 64
cache_alignment	: 64
address sizes	: 40 bits physical, 48 bits virtual
power management:


cores 1-7: same lines with minor change in apicid lines.

**Comment by LuggyLuke** [comment]
well after initially finding your post on the forum and thinking, nah i have a semi recent cpu, that cant be... two days later i come back and check again. So not only old cpus dont have AVX. also semi recent budget cpus (G4560T) might not support it. Just in case someone else is to stuck up to admit it might be the cpu XD. Googling around i found this repo supplying a 1.8 build for non AVX, it wont install on my machine, but maybe u have more luck. 
https://github.com/yaroslavvb/tensorflow-community-wheels/issues/68

anyways i hope there will be some way to get this working again, i rly dont like pocketsphinx.

**Comment by LuggyLuke** [comment]
Ah, thanks. i might look into that, but how would i build a precise release with the modified tf? Is that possible as well?

**Comment by el-tocino** [comment]
Remove the existing tf package (pip3 uninstall tensorflow), then build the new tf wheel per the instructions (https://www.tensorflow.org/install/source) and install it (pip3 install xyz). Precise probably works with the current version, but you can clone the previous version sources and build those if necessary as well.

**Comment by LuggyLuke** [comment]
Ah well i just installed TF 1.5 and it seems to work. was stuck on creating the tar.gz file and using it with openhab. got it working now (just run the build.sh in case some other newb like me wants to do it too). So for now precise seems to work like a charm, but i will need to do some long term testing. sadly the new hotword (hey-mycroft-2) cant be used, but the old one still does the job. Thank you guys for this and also the fast response/support :+1:

**Comment by daniele777** [comment]
Guys i have same problem illegal istruction i tried with TensorFlow 2.0 but no changed ... 
i have CPU intel core i7 920 .... how to fix????

**Comment by el-tocino** [comment]
Read Matthew's reply above; you'd have to compile your own tensorflow for a cpu that old.

**Comment by daniele777** [comment]
here can i make?  git checkout branch_name  # r2.2, r2.3, etc. good versione of TensorFlow ?

**Comment by el-tocino** [comment]
google compiling tensorflow, there's a good set of docs on it.

**Comment by el-tocino** [comment]
you'd want the relevant version specified for the branch you're using. 1.13 should work for most things, otherwis check setup.py to see.

**Comment by daniele777** [comment]
can i use amd for gpu support? MROC?
